libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -811,23 +803,31 @@ static DNNReturnType execute_model_tf(const DNNModel *model, const char *input_n
         return DNN_ERROR;
     }
 
+    output = av_malloc_array(nb_output, sizeof(*output));
     for (uint32_t i = 0; i < nb_output; ++i) {
-        output.height = TF_Dim(output_tensors[i], 1);
-        output.width = TF_Dim(output_tensors[i], 2);
-        output.channels = TF_Dim(output_tensors[i], 3);
-        output.data = TF_TensorData(output_tensors[i]);
-        output.dt = TF_TensorType(output_tensors[i]);
-
+        output[i].height = TF_Dim(output_tensors[i], 1);
+        output[i].width = TF_Dim(output_tensors[i], 2);
+        output[i].channels = TF_Dim(output_tensors[i], 3);
+        output[i].data = TF_TensorData(output_tensors[i]);
+        output[i].dt = TF_TensorType(output_tensors[i]);
+    }
+    switch (model->func_type) {
```

Comment:
changes from line 814 should not be in this patch (special for multiple output support), you might add a new patch special for detection support in tf backend if needed.